teps/0138-decouple-api-and-feature-versioning.md

```diff
@@ -72,8 +85,116 @@ Per [TEP-0033](https://github.com/tektoncd/community/blob/main/teps/0033-tekton-
 **Tekton Maintainers**
 - I would like to be able to migrate the apiVersion without having to make backwards incompatible changes.
 
+## Proposal
+The solutions and alternatives aim to decouple api and feature versioning, and will provide a migration plan for setting the default value of `enable-api-fields` back to `stable` after the notification window for 9 months. Note that with all solutions proposed, this is a breaking change since when we change `enable-api-fields` to `stable`, all `beta` features will go from enabled by default to disabled by default. However, some solutions will provide workarounds to help with the transition back to `stable`.
+
+## Design Details
+### Introduce “new-stable” value for enable-api-fields
+
+- A new option, `enable-api-fields` = `new-stable`, will be added to the API. This option will use the preferred validation where `stable` and `beta` features are validated the same across apiVersions. In 9 months, `new-stable` will be renamed as `stable`.
+- The current behavior for “stable” `enable-api-fields` remains the same for now where it allows users to use `beta` features in `v1beta1` that have coupled feature and api versioning in `v1beta1` i.e. remote resolution. In 9 months “stable” will be renamed as “legacy-stable” and be marked as deprecated.
+- The existing `beta` option will remain the same for now and in 9 months.
+
+- Taking the remote resolution feature, which currently couples feature and api versioning, as an example, after the change:
+  - With `new-stable` in `v1beta1`, we disable resolver.
+  - With `legacy-stable` in `v1beta1`, we do not validate resolvers as a field needs `enable-api-fields` as beta, so they are still turned on by default.
+  - When `enable-api-fields` is set to `beta` in both `v1` and `v1beta1`, we are turning on resolver.
+
+### Provide an immediate solution to decouple api and feature versioning now
+This solution proposes that we provide an immediate solution to turn on stable features only with `stable` `enable-api-fields` and require `enable-api-fields` to be set to `beta` when using beta features, regardless of CRD API version. This will solve the unintended behavior that taskruns cannot be created for version coupled features with “enable-api-fields” set to “stable” during the [v1 storage swap](https://github.com/tektoncd/pipeline/pull/6444#issuecomment-1580926707) right away.
```

Comment:
Can you reword this sentence a little bit? This sentence makes it sound like the behavior of "enable-api-fields=stable" is changing immediately, but the "stable" value isn't changing for 9 months. What's changing now is that we're providing a new value to address the problem.

Reply:
Thanks, that makes sense! Updated.